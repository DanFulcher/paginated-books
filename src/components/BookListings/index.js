import React from 'react';
import PropTypes from 'prop-types';
import {Row, Col, Card} from 'react-bootstrap';

const BookListings = ({books}) => {
  return (
    <Row>
      {books && books.map(book => (
        <Col key={book.id} xs={12} sm={6} md={4} className="mb-4">
          <Card key={`book-${book.id}`} className="w-100 h-100">
            <Card.Body>
              <Card.Title>{book.book_title}</Card.Title>
              <Card.Text>
                {book.book_author.map(bookAuthor => bookAuthor)}
              </Card.Text>
              <Card.Text>Book pages: {book.book_pages}</Card.Text>
              <Card.Text>
      Publication in: {book.book_publication_city}, {book.book_publication_country}({book.book_publication_year})
              </Card.Text>
            </Card.Body>
          </Card>
        </Col>
      ))}
    </Row>
  );
};
BookListings.propTypes = {
  books: PropTypes.arrayOf(PropTypes.object),
}

export default BookListings;
