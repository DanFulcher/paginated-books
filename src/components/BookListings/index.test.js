import React from 'react';
import BookListings from './index';
import renderer from 'react-test-renderer';

const data = [{
  book_title: 'Test Book',
  book_author: ["Test Author"],
  book_pages: 123,
  book_publication_city: 'Test City',
  book_publication_country: 'Test Country',
  book_publication_year: 1234,
  id: 2468
}]

test('Correctly Renders Book Listing component', () => {
  const component = renderer.create(
    <BookListings books={data} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});