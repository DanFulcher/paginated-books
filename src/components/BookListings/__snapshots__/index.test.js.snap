// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly Renders Book Listing component 1`] = `
<div
  className="row"
>
  <div
    className="mb-4 col-md-4 col-sm-6 col-12"
  >
    <div
      className="w-100 h-100 card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Test Book
        </div>
        <p
          className="card-text"
        >
          Test Author
        </p>
        <p
          className="card-text"
        >
          Book pages: 
          123
        </p>
        <p
          className="card-text"
        >
          Published in: 
          Test City
          , 
          Test Country
           (
          1234
          )
        </p>
      </div>
    </div>
  </div>
</div>
`;
